server/api/users/users.controller.ts

```diff
@@ -49,6 +49,10 @@ export const getUserById: RequestHandler<{ id: string }, UserProfile> = async (r
 
 export const updateUserById: RequestHandler<{ id: string }, string, UpdatableFields> = async (req, res) => {
 
+  if (Object.keys(req.body).length === 0) {
+    throw new BadRequestError('Request body should not be an empty object.');
+  }
```

Comment:
shouldn't we instead be checking for `toUpdate`? since they could have a req.body.asdf which wouldn't get picked by the update object, but wouldn't be caught here and would result in a no-op update later